scalopus_tracing/test/test_native_tracepoints.cpp

```diff
@@ -202,5 +200,22 @@ int main(int /* argc */, char** /* argv */)
   test(result[0]["name"], "counter");
   test(result[0]["ph"], "C");
 
+  // Ensure we don't lose events if thread goes out of scope immediately
+  source->startInterval();
+  std::this_thread::sleep_for(std::chrono::milliseconds(100));
+  // From another thread emit a tracepoint.
+  std::thread immediately_closing_thread = std::thread([]() {
+    {
+      TRACE_SCOPE_RAII("immediately_closing_thread");
+    }
+  });
```

Comment:
This unit test without the sleep at first failed, with this branch it now passes.

Reply:
Not sure I'm following you here. Do you still need the sleep?

Reply:
Oh, sorry, it's worded poorly.

We don't need the sleep anymore. What I meant is that I added this new unit test, that didn't have any sleeps, that allowed me to confirm it failed with the old code. Then I wrote the orphaned buffer and ensured this passed. Then I discovered that  one of the other unit tests had a sleep before the thread went out of scope, where I discovered that I apparently had already thought of this failure mode, and since the failure is now prevented I removed the sleep there as well.

Does this clear it up?

Reply:
I was confused because there are some other sleeps, but they're for a different purpose. LGTM!

This is a great fix. :smiley: